### PR TITLE
bv2int: new options for bvand translation

### PR DESCRIPTION
**Pull request by yoni206**
Currently, `(bvand x y)` is translated into a sum of ITEs. 
This PR introduces two more options for the translation of `(bvand x y)`:
1. bv: cast the integer translations of `x` and `y` back to bit-vectors, do a `bvand`, and cast the result to integers.
2. iand: use the builtin `iand` operator.

These options are added to many of the tests, and some new tests are added.
In addition, some tests are cleaned up (e.g., removing --no-check-unsat-cores for satisfiable benchmarks).
Finally, some tests are moved from regress0 to regress2 because they take several seconds to complete (2 -- 10 seconds).

### CODE REVIEW

File: test/regress/regress3/bv_to_int_check_ne_bvlshr0_4bit.smt2 [hunk 1]
```diff
@@ -0,0 +1,19 @@
+; COMMAND-LINE:  --solve-bv-as-int=bv 
+(set-logic BV)
```
Comment (ajreynol) [inline test/regress/regress3/bv_to_int_check_ne_bvlshr0_4bit.smt2:2]: Seems to be missing a status.

Comment (ajreynol) [reply test/regress/regress3/bv_to_int_check_ne_bvlshr0_4bit.smt2:2]: Note that regress3 is not checked by CI, so you should maybe ensure `make regress3` works locally.

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: test/regress/regress3/bv_to_int_check_ne_bvlshr0_4bit.smt2 (hunk 1 above)
Comment (yoni206) [reply test/regress/regress3/bv_to_int_check_ne_bvlshr0_4bit.smt2]: this file was actually redundant and did not appear in CMakeLists. fixed now. `ctest -R bv_to_int` passes locally.

**Review by ajreynol — APPROVED** [review]
LGTM

**Pull request merged**